node/hyperbahn/service_proxy.js

```diff
@@ -89,6 +89,58 @@ util.inherits(ServiceDispatchHandler, EventEmitter);
 
 ServiceDispatchHandler.prototype.type = 'tchannel.hyperbahn.service-dispatch-handler';
 
+ServiceDispatchHandler.prototype.handleLazily =
+function handleLazily(conn, reqFrame) {
+    var self = this;
+
+    // TODO: this code and _sendLazyErrorFrame are nearly identical to
+    // counterparts in TChanneLServiceNameHandler, sharing would be preferable
+
+    var res = reqFrame.bodyRW.lazy.readService(reqFrame);
+    if (res.err) {
+        // TODO: stat?
+        self.channel.logger.warn('failed to lazy read frame serviceName', conn.extendLogInfo({
+            error: res.err
+        }));
+        // TODO: protocol error instead?
+        self._sendLazyErrorFrame(conn, reqFrame, 'BadRequest', 'failed to read serviceName');
+        return false;
+    }
+
+    var serviceName = res.value;
+    if (!serviceName) {
+        // TODO: reqFrame.extendLogInfo would be nice, especially if it added
+        // things like callerName and arg1
+        self.channel.logger.warn('missing service name in lazy frame', conn.extendLogInfo({}));
+        self._sendLazyErrorFrame(conn, reqFrame, 'BadRequest', 'missing serviceName');
+        return false;
+    }
+
+    var chan = self.channel.subChannels[serviceName];
+    if (!chan) {
+        chan = self.getOrCreateServiceChannel(serviceName);
```

Comment:
This is a change in semantics. We never call this method in `handlRequest`.

Consider returning `false` here